src/bicep/modules/privateLink.bicep

```diff
@@ -0,0 +1,211 @@
+@description('The name of the resource the private endpoint is being created for')
+param laWorkspaceName string
+
+@description('The resource id of the resoure the private endpoint is being created for')
+param laWorkspaceResourceId string
+
+@description('The region the private endpoint will be created in')
+param privateEndpointLocation string = resourceGroup().location
+
+@description('The name of the subnet in the virtual network where the private endpoint will be placed')
+param privateEndpointSubnetName string
+
+@description('The name of the virtual network where the private endpoint will be placed')
+param privateEndpointVnetName string
+
+@description('The tags that will be associated to the VM')
+param tags object
+
+@description('Data used to append to resources to ensure uniqueness')
+param uniqueData string = substring(newGuid(), 0, 8)
```

Comment:
should be a `var`? or if we wanted it a parameter, to keep modules unopinionated, be passed by the calling module with a [bicep `take()`](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-array#take) call on the unique string generated at the root module?

Reply:
Its a parameter only because you can use newGuid() in variable :( but also if running stand alone and a user wants to use their own. Not using passed from root since it uniqueness is different and I had challenging conflict results not working when using take from root. Will consider better option in future but since naming was a root of many issues in testing and this worked ;)